src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -196,7 +213,7 @@ public void ICollection_NonGeneric_SyncRoot_MatchesExpectedType(int count)
         [MemberData(nameof(ValidCollectionSizes))]
         public void ICollection_NonGeneric_SyncRoot_ThrowsNotSupportedException(int count)
         {
-            if (!ICollection_NonGeneric_SupportsSyncRoot)
+            if (!ICollection_NonGeneric_SupportsSyncRoot && !ICollection_NonGeneric_HasNullSyncRoot)
```

Comment:
Rather than having this be a separate test, I'd similarly rather have it be in the single test that verifies the SyncRoot is what we expect it to be: null, non-null, or not even supported.  I realize you didn't introduce this particular test.  I'm just cognizant of the fact that every new test we add here gets multiplied significantly, as it's a theory that's then pulled into a bunch of different tests for a bunch of different instantiations for a bunch of different collection types.  If a test really is verifying something unique, then it's fine to stand alone.  Here, though, this and the other two earlier tests are really all checking the same thing: does SyncRoot return what we expect (if we expect anything).